src/client/game.cpp

```diff
@@ -2977,22 +2982,21 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 	core::line3d<f32> shootline;
 
 	if (camera->getCameraMode() != CAMERA_MODE_THIRD_FRONT) {
-		shootline = core::line3d<f32>(camera_position,
-			camera_position + camera_direction * BS * d);
+		shootline = core::line3d<f32>(player_eye_position,
+			player_eye_position + camera_direction * BS * d);
 	} else {
 	    // prevent player pointing anything in front-view
-		shootline = core::line3d<f32>(camera_position,camera_position);
+		shootline = core::line3d<f32>(camera_position, camera_position);
 	}
 
 #ifdef HAVE_TOUCHSCREENGUI
 
 	if ((g_settings->getBool("touchtarget")) && (g_touchscreengui)) {
 		shootline = g_touchscreengui->getShootline();
 		// Scale shootline to the acual distance the player can reach
-		shootline.end = shootline.start
```

Comment:
`shootline.start` is needed here so that mobile players without the crosshair mode can point at the correct position.

Reply:
Will try.

Reply:
Done. This fix is inapplicable for non-crosshair input.